src/test/java/io/cdap/plugin/couchbase/etl/BaseCouchbaseETLTest.java

```diff
@@ -109,36 +103,15 @@ public static void setupTestClass() throws Exception {
       .build();
 
     clusterManager.insertBucket(bucketSettings);
-    bucket = cluster.openBucket(bucketName);
+    Bucket bucket = cluster.openBucket(bucketName);
     // Buckets with no index cannot be queried. Documents can only be retrieved by making use of the USE KEYS operator
-    bucket.bucketManager().createN1qlPrimaryIndex(true, false, 60, TimeUnit.SECONDS);
-  }
-
-  public List<StructuredRecord> getPipelineResults(Map<String, String> sourceProperties) throws Exception {
-    Map<String, String> allProperties = new ImmutableMap.Builder<String, String>()
-      .put(Constants.Reference.REFERENCE_NAME, name.getMethodName())
-      .putAll(sourceProperties)
-      .build();
+    bucket.bucketManager().createN1qlPrimaryIndex(true, false, COUCHBASE_TIMEOUT, TimeUnit.SECONDS);
 
-    ETLStage source = new ETLStage("CouchbaseSource", new ETLPlugin(CouchbaseConstants.PLUGIN_NAME,
-                                                                    BatchSource.PLUGIN_TYPE, allProperties, null));
-
-    String outputDatasetName = "output-batchsourcetest_" + name.getMethodName();
-    ETLStage sink = new ETLStage("sink", MockSink.getPlugin(outputDatasetName));
-
-    ETLBatchConfig etlConfig = ETLBatchConfig.builder()
-      .addStage(source)
-      .addStage(sink)
-      .addConnection(source.getName(), sink.getName())
-      .build();
-
-    ApplicationId pipelineId = NamespaceId.DEFAULT.app("Couchbase_" + name.getMethodName());
-    ApplicationManager appManager = deployApplication(pipelineId, new AppRequest<>(APP_ARTIFACT, etlConfig));
-
-    WorkflowManager workflowManager = appManager.getWorkflowManager(SmartWorkflow.NAME);
-    workflowManager.startAndWaitForRun(ProgramRunStatus.COMPLETED, 5, TimeUnit.MINUTES);
+    return bucket;
+  }
 
-    DataSetManager<Table> outputManager = getDataset(outputDatasetName);
-    return MockSink.readOutput(outputManager);
+  @AfterClass
+  public static void afterTestClass() throws Exception {
+    bucket.close();
```

Comment:
shouldn't we also close a cluster?

Reply:
Yes. I missed `cluster.disconnect()` method invocation. Added it to tests and also to the record reader and to the record writer. Thanks.